### PR TITLE
Qb 573 potrewardsbyownerandheight

### PR DESCRIPTION
**Pull request by hong-pang**
1. removed getPotRewardsByOwnerHandlerFn in pot/client/rest/query.go
2. added getPotRewardsHandlerFn in  pot/client/rest/query.go
3. change if..else statements to switch in register/keeper/querier.go

### CODE REVIEW

File: x/pot/client/rest/query.go [hunk 1]
```diff
@@ -14,7 +14,8 @@ import (
 
 func registerQueryRoutes(cliCtx context.CLIContext, r *mux.Router) {
 	r.HandleFunc("/pot/rewards/epoch/{epoch}", getPotRewardsByEpochHandlerFn(cliCtx, keeper.QueryPotRewardsByEpoch)).Methods("GET")
-	r.HandleFunc("/pot/rewards/owner/{ownerAddress}", getPotRewardsByOwnerHandlerFn(cliCtx, keeper.QueryPotRewardsByOwner)).Methods("GET")
+	//r.HandleFunc("/pot/rewards/owner/{ownerAddress}", getPotRewardsByOwnerHandlerFn(cliCtx, keeper.QueryPotRewardsByOwner)).Methods("GET")
```
Comment (alexstratos) [inline x/pot/client/rest/query.go:17]: why this records is commented instead removed?

File: x/pot/client/rest/query.go [hunk 2]
```diff
@@ -86,65 +122,83 @@ func getPotRewardsByOwnerHandlerFn(cliCtx context.CLIContext, queryPath string)
...
-
-func checkEpoch(w http.ResponseWriter, r *http.Request, epochStr string) (sdk.Int, bool) {
-	epoch, ok := sdk.NewIntFromString(epochStr)
-	if !ok {
-		rest.WriteErrorResponse(w, http.StatusBadRequest, "Invalid 'epoch'.")
-		return sdk.NewInt(0), false
-	}
-	return epoch, true
-}
+//func getPotRewardsHandlerFn(cliCtx context.CLIContext, queryPath string) http.HandlerFunc {
```
Comment (alexstratos) [inline x/pot/client/rest/query.go:158]: why keep a copy of the same function?

File: x/pot/keeper/distribute.go [hunk 3]
```diff
@@ -6,6 +6,8 @@ import (
 )
 
 func (k Keeper) DistributePotReward(ctx sdk.Context, trafficList []types.SingleNodeVolume, epoch sdk.Int) (totalConsumedOzone sdk.Dec, err error) {
+	//nodeOwnerMap := k.getNodeOwnerMap(ctx)
```
Comment (alexstratos) [inline x/pot/keeper/distribute.go:9]: why keep the commented code?

File: x/pot/keeper/querier.go [hunk 4]
```diff
@@ -24,12 +23,14 @@ func NewQuerier(k Keeper) sdk.Querier {
...
 		case QueryVolumeReport:
 			return queryVolumeReport(ctx, req, k)
-		case QueryPotRewards:
-			return queryPotRewards(ctx, req, k)
+		//case QueryPotRewards:
+		//	return queryPotRewards(ctx, req, k)
 		case QueryPotRewardsByEpoch:
 			return queryPotRewardsByEpoch(ctx, req, k)
+		//case QueryPotRewardsByOwner:
+		//	return queryPotRewardsByOwner(ctx, req, k)
```
Comment (alexstratos) [inline x/pot/keeper/querier.go:31]: why keep duplicated case commented

File: x/pot/client/rest/query.go [hunk 5]
```diff
@@ -86,65 +122,83 @@ func getPotRewardsByOwnerHandlerFn(cliCtx context.CLIContext, queryPath string)
...
 		route := fmt.Sprintf("custom/%s/%s", types.QuerierRoute, queryPath)
-		res, height, err := cliCtx.QueryWithData(route, bz)
+		res, _, err := cliCtx.QueryWithData(route, bz)
 		if err != nil {
 			rest.WriteErrorResponse(w, http.StatusInternalServerError, err.Error())
 			return
 		}
 
-		cliCtx = cliCtx.WithHeight(height)
+		//cliCtx = cliCtx.WithHeight(height)
```
Comment (alexstratos) [inline x/pot/client/rest/query.go:153]: why keep commented code here?

File: x/pot/keeper/distribute.go [hunk 6]
```diff
@@ -267,9 +272,14 @@ func (k Keeper) addNewRewardAndReCalcTotal(ctx sdk.Context, account sdk.AccAddre
 		k.setRewardAddressPool(ctx, rewardAddressPool)
 	}
 
+	distributionRecord := NewNodeRewardsInfo(account, matureTotal, immatureTotal)
+	potRewardsRecordVal := NewNodeRewardsRecord(ctx.BlockHeight(), currentEpoch, distributionRecord)
+
 	k.setMatureTotalReward(ctx, account, matureTotal)
 	k.setImmatureTotalReward(ctx, account, immatureTotal)
 	k.setIndividualReward(ctx, account, matureEpoch, newReward)
+	//k.setPotRewardRecord(ctx, currentEpoch, potRewardsRecordVal)
```
Comment (alexstratos) [inline x/pot/keeper/distribute.go:281]: why keep commented code

File: x/pot/keeper/querier.go [hunk 7]
```diff
@@ -209,3 +113,38 @@ func getFilteredNodesAddrByOwner(ctx sdk.Context, ownerAddress sdk.AccAddress, k
...
+
+func queryPotRewardsWithOwnerHeight(ctx sdk.Context, req abci.RequestQuery, k Keeper) ([]byte, error) {
+	var params QueryPotRewardsWithOwnerHeightParams
+	err := k.cdc.UnmarshalJSON(req.Data, &params)
+	if err != nil {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrJSONMarshal, err.Error())
+	}
+	if err != nil {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrJSONMarshal, err.Error())
+	}
```
Comment (alexstratos) [inline x/pot/keeper/querier.go:125]: duplicated code

File: x/pot/types/key.go [hunk 8]
```diff
@@ -31,7 +31,8 @@ var (
 	MatureTotalRewardKeyPrefix   = []byte{0x14} // key: prefix{address}_mature_total
 	ImmatureTotalRewardKeyPrefix = []byte{0x15} // key: prefix{address}_immature_total
 
-	EpochRewardsKeyPrefix = []byte{0x51} // key: prefix{epoch}, value: rewardDetailMap (map[node_address string]types.Reward)
+	EpochRewardsKeyPrefix     = []byte{0x51} // key: prefix{epoch}, value: []types.Reward
```
Comment (alexstratos) [inline x/pot/types/key.go:34]: I see now other place is using it.

File: x/pot/client/rest/query.go [hunk 9]
```diff
@@ -66,8 +67,43 @@ func getPotRewardsByEpochHandlerFn(cliCtx context.CLIContext, queryPath string)
...
+		cliCtx, ok := rest.ParseQueryHeightOrReturnBadRequest(w, cliCtx, r)
+		if !ok {
+			return
+		}
+
+		v := mux.Vars(r)["epoch"]
+		epoch, ok := checkEpoch(w, r, v)
+		if len(v) == 0 || !ok {
+			return
+		}
```
[the remark this answers is not in the document]
Comment (hong-pang) [reply x/pot/client/rest/query.go:82]: v is still a string here

File: x/pot/client/rest/query.go (hunk 5 above)
Comment (hong-pang) [reply x/pot/client/rest/query.go:153]: removed comment

File: x/pot/client/rest/query.go (hunk 2 above)
Comment (hong-pang) [reply x/pot/client/rest/query.go:158]: removed

File: x/pot/keeper/distribute.go (hunk 3 above)
Comment (hong-pang) [reply x/pot/keeper/distribute.go:9]: removed comment

File: x/pot/keeper/distribute.go [hunk 10]
```diff
@@ -228,18 +232,19 @@ func (k Keeper) CalcMiningRewardInTotal(ctx sdk.Context, distributeGoal types.Di
...
 	matureEpoch := k.getMatureEpochByCurrentEpoch(ctx, currentEpoch)
 	for _, reward := range rewardDetailMap {
 		nodeAddr := reward.NodeAddress
 		totalReward := reward.RewardFromMiningPool.Add(reward.RewardFromTrafficPool)
-		k.addNewRewardAndReCalcTotal(ctx, nodeAddr, currentEpoch, matureEpoch, totalReward)
+		res = append(res, k.addNewRewardAndReCalcTotal(ctx, nodeAddr, currentEpoch, matureEpoch, totalReward))
 	}
 	k.setLastMaturedEpoch(ctx, currentEpoch)
-	return nil
+	//k.setPotRewardRecord(ctx, currentEpoch, res)
```
[the remark this answers is not in the document]
Comment (hong-pang) [reply x/pot/keeper/distribute.go:243]: removed comment

File: x/pot/keeper/distribute.go (hunk 6 above)
Comment (hong-pang) [reply x/pot/keeper/distribute.go:281]: removed comment

File: x/pot/keeper/querier.go (hunk 4 above)
Comment (hong-pang) [reply x/pot/keeper/querier.go:31]: removed comment

File: x/pot/keeper/querier.go (hunk 7 above)
Comment (hong-pang) [reply x/pot/keeper/querier.go:125]: removed duplicates

File: x/pot/types/key.go (hunk 8 above)
Comment (hong-pang) [reply x/pot/types/key.go:34]: removed EpochRewardsKeyPrefix. No use any more.

File: x/pot/client/rest/query.go [hunk 11]
```diff
@@ -66,8 +67,43 @@ func getPotRewardsByEpochHandlerFn(cliCtx context.CLIContext, queryPath string)
...
+		cliCtx = cliCtx.WithHeight(height)
+		rest.PostProcessResponse(w, cliCtx, res)
+	}
+}
+
+func checkEpoch(w http.ResponseWriter, r *http.Request, epochStr string) (sdk.Int, bool) {
+	epoch, ok := sdk.NewIntFromString(epochStr)
+	if !ok {
+		rest.WriteErrorResponse(w, http.StatusBadRequest, "Invalid 'epoch'.")
+		return sdk.NewInt(0), false
```
[the remark this answers is not in the document]
Comment (hong-pang) [reply x/pot/client/rest/query.go]: set epoch as -1

**Pull request merged**